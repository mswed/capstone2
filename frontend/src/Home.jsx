import { useState, useEffect } from 'react';
import GrumpyApi from './api';

const Home = () => {
  const [stats, setStats] = useState({});

  useEffect(() => {
    async function getStats() {
      try {
        const res = await GrumpyApi.getStats();
        setStats(res);
      } catch (error) {
        console.error('Error fetching stats', error);
      }
    }
    getStats();
  }, [stats]);

  return (
    <div>
      <h1>Welcome to the Grumpy Tracker!</h1>
      <h3>{stats.projects} Projects</h3>
      <h3>{stats.makes} Makes</h3>
      <h3>{stats.cameras} Cameras</h3>
      <h3>{stats.formats} Formats</h3>
      <h3>Endless complaints... </h3>
    </div>
  );
};

export default Home;
