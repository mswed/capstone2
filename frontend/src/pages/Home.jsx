import { useState, useEffect } from 'react';
import { Container, Row, Col } from 'react-bootstrap';
import GrumpyApi from '../services/api';

const Home = () => {
  const [stats, setStats] = useState({});

  useEffect(() => {
    async function getStats() {
      try {
        const res = await GrumpyApi.getStats();
        setStats(res);
      } catch (error) {
        console.error('Error fetching stats', error);
      }
    }
    getStats();
  }, []);

  return (
    <Container className="mt-3">
      <Row>
        <Col>
          <div
            style={{
              backgroundImage: 'url("/home.png")',
              backgroundSize: 'cover',
              backgroundPosition: 'center',
              backgroundRepeat: 'no-repeat',
              minHeight: '400px',
              position: 'relative',
              color: 'white',
              textShadow: '2px 2px 4px rgba(0, 0, 0, 0.8)',
            }}
          >
            <h3 style={{ position: 'absolute', top: '20px', left: '20px' }}>{stats.projects} Projects</h3>

            <h3 style={{ position: 'absolute', top: '80px', left: '120px' }}>{stats.makes} Makes</h3>

            <h3 style={{ position: 'absolute', top: '160px', left: '200px' }}>{stats.cameras} Cameras</h3>

            <h3 style={{ position: 'absolute', top: '240px', left: '280px' }}>{stats.formats} Formats</h3>

            <h3 style={{ position: 'absolute', top: '320px', left: '360px' }}>Endless complaints...</h3>
          </div>
        </Col>
      </Row>
    </Container>
  );
};

export default Home;
