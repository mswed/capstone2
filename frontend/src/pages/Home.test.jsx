import { vi, it, expect, beforeEach } from 'vitest';
import { render, screen, waitFor } from '@testing-library/react';
import Home from './Home';

// Mock API calls
vi.mock('../services/api', () => ({
  default: {
    getStats: vi.fn(() =>
      Promise.resolve({
        projects: 30,
        cameras: 10,
        makes: 3,
        formats: 200,
      })
    ),
  },
}));

describe('Home component', () => {
  beforeEach(() => {
    // Clear mocks
    vi.clearAllMocks();
  });

  it('renders without crashing', async () => {
    render(<Home />);
    await screen.findByText('30 Projects');
  });

  it('matches snapshot', async () => {
    const { asFragment } = render(<Home />);

    await screen.findByText('30 Projects');
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays the correct stats', async () => {
    render(<Home />);

    await waitFor(() => {
      expect(screen.getByText('30 Projects')).toBeInTheDocument();
      expect(screen.getByText('3 Makes')).toBeInTheDocument();
      expect(screen.getByText('10 Cameras')).toBeInTheDocument();
      expect(screen.getByText('200 Formats')).toBeInTheDocument();
      expect(screen.getByText('Endless complaints...')).toBeInTheDocument();
    });
  });
});
