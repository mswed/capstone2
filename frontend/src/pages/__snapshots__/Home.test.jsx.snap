// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="mt-3 container"
  >
    <h3>
      30 Projects
    </h3>
    <h3>
      3 Makes
    </h3>
    <h3>
      10 Cameras
    </h3>
    <h3>
      200 Formats
    </h3>
    <h3>
      Endless complaints... 
    </h3>
  </div>
</DocumentFragment>
`;
